src/main/java/io/lettuce/core/cluster/models/partitions/RedisClusterNode.java

```diff
@@ -103,8 +103,7 @@ public RedisClusterNode(RedisURI uri, String nodeId, boolean connected, String s
         this.configEpoch = configEpoch;
         this.replOffset = -1;
 
-        this.slots = new BitSet(slots.length());
-        this.slots.or(slots);
+        this.slots = slots != null ? slots : new BitSet(0);
```

Comment:
I think part of the idea behind the existing code is to clone the content of the slots, instead of directly assigning the provided value.

This has the added benefit of isolation, because the code that called the constructor could no longer modify the contents of the slot (as a side effect).

However you are correct that a good design should consider this method receiving a null for the slots argument. Perhaps something in the line of:

```java
this.slots = new BitSet(SlotHash.SLOT_COUNT);
this.slots.or(slots);
```

This way:
- if the slots argument are null then the `or()` method handles it correctly
- the `this.slots` is always initialized with an empty list of slots

IMHO the consistency of always having BitSet initialized with the `SlotHash.SLOT_COUNT` pays off for the "waste" of memory.

Reply:
@tishun Thank you for your feedback. I have made the suggested changes.